vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -92,69 +107,48 @@ public ImageData getImageDimensions(IVSImage image, int width, int height) {
      * @return list of images in the requested pageNo
      */
     @Override
-    public List<IVSImage> getImages(int pageNo) {
-        return getImages(pageNo, SortByField.CREATION_DATE.getValue(), Sort.Direction.DESC.toString());
+    public List<IVSImage> getImages(int pageNo, String category) {
+        return getImages(pageNo, category, SortByField.CREATION_DATE.getValue(), Sort.Direction.DESC.toString());
     }
 
-    /**
-     * Method to return the requested images
-     * 
-     * @param pageNo. if pageNo<1, 1st page is returned, if pageNo>total pages,last
-     *                page is returned
-     * @return list of images in the requested pageNo and requested order.
-     */
     @Override
-    public List<IVSImage> getImages(int pageNo, String sortedBy, String order) {
-        Sort sortingParameters = Sort.by(SortByField.CREATION_DATE.getValue()).descending();
-        pageNo = validatePageNumber(pageNo);
-        if(sortedBy!=null && SortByField.getAllValues().contains(sortedBy)) {
-            sortingParameters = Sort.by(sortedBy);
-        }
-        if(order!=null && order.equalsIgnoreCase(Sort.Direction.ASC.toString())) {
-            sortingParameters = sortingParameters.ascending();
-        }
-        else {
-            sortingParameters = sortingParameters.descending();
-        }
+    public List<IVSImage> getImages(int pageNo, String category, String sortedBy, String order) {
```

Comment:
can't you just use `null` to get all categories? It would be more intuitive than a string that in theory could be anything.

Reply:
In that case, URL query parameter value for imageCat would also be shown as null?

Reply:
these are two separate things. You can pass an empty query parameter but pass null into the service class.